File: src/test/java/io/github/subiyacryolite/jds/entities/EntityB.java
package io.github.subiyacryolite.jds.entities;

import io.github.subiyacryolite.jds.JdsField;
import io.github.subiyacryolite.jds.annotations.JdsEntityAnnotation;
import io.github.subiyacryolite.jds.enums.JdsFieldType;
import javafx.beans.property.SimpleStringProperty;

/**
 * Created by ifunga on 01/07/2017.
 */
@JdsEntityAnnotation(entityId = 1001, entityName = "entityb")
public class EntityB extends EntityA {
    private static final JdsField ENTITY_B_FIELD = new JdsField(5001, "entity_b_field", JdsFieldType.TEXT);
    private final SimpleStringProperty field = new SimpleStringProperty("C");

    public EntityB() {
        map(ENTITY_B_FIELD, field);
    }

    public void setEntityBValue(String s) {
        field.set(s);
    }

    public String getEntityBValue() {
        return field.get();
    }

    @Override
    public String toString() {
        return "EntityB{" +
                "field A = " + getEntityAValue() +
                ",field B = " + getEntityBValue() +
                '}';
    }
}

File: src/test/java/entities/EntityB.kt
package entities

import io.github.subiyacryolite.jds.JdsField
import io.github.subiyacryolite.jds.annotations.JdsEntityAnnotation
import io.github.subiyacryolite.jds.enums.JdsFieldType
import javafx.beans.property.SimpleStringProperty

/**
 * Created by ifunga on 01/07/2017.
 */
@JdsEntityAnnotation(id = 1001, name = "entityb")
open class EntityB : EntityA() {
    private val _field = SimpleStringProperty("C")

    init {
        map(ENTITY_B_FIELD, _field)
    }

    var entityBValue: String
        get() = _field.get()
        set(s) = _field.set(s)

    override fun toString(): String {
        return "{ EntityB { FieldA = $entityAValue, FieldB = $entityBValue } }"
    }

    companion object {
        private val ENTITY_B_FIELD = JdsField(5001, "entity_b_field", JdsFieldType.STRING)
    }
}
